js/lib/baseDomain.js

```diff
@@ -9,14 +9,27 @@ const LRUCache = require('lru_cache/core').LRUCache
 
 let cachedBaseDomain = new LRUCache(50)
 
+const checkASCII = function (str) {
+  if (typeof str !== 'string') {
+    return false
+  }
+  for (let i = 0; i < str.length; i++) {
+    if (str.charCodeAt(i) > 127) {
```

Comment:
I think the impl is probably right though in regards to what I brought up because you will just return false before it gets to the second code unit, so you can probably ignore my comment above.